duckduckgo/iOS
# Nightly alpha builds

Task/Issue URL: https://app.asana.com/0/0/1204986042798452/f
Tech Design URL: https://app.asana.com/0/481882893211075/1204986042798431/f
CC: @loremattei @THISISDINOSAUR @ayoy 

**Description**:
Right now, we’re concentrating most of our internal testing on the release candidates. This means that we have a lot fewer eyes on issues that might have been merged much earlier in the cycle. Furthermore, we have to rely on bespoke builds for anything that’s kept only do the Debug build config and rely on people needing to authenticate to see debug options (which means only the latest OS versions).

The ability to build alphas was already merged in #1913 but it still relied on building an distributing locally from Xcode. This PR adds:
- The ability to build Alphas to a specified TestFlight group from GHA
- Nightly Alpha build deployments

**Steps to test this PR**:
1. Take a look at the [most recent successful build](https://github.com/duckduckgo/iOS/actions/runs/6170933936/job/16748187934) from the added action
2. Take a look the DuckDuckGo Alpha TestFlight on App Store Connect and check that the build number from the recent build’s logs is visible there. It’s been distributed to the "Latest Alpha Group” for testing / outside of nightly deployment purposes. Feel free to add yourself to this group and check that you see the build on TestFlight (but that’s just testing TestFlight so probably not necessary).
3. Check that the Alpha build dSyms have been uploaded to the [Asana task](https://app.asana.com/0/0/1205344386326139/f)

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
